src/ralph/assets/models/components.py

```diff
@@ -177,3 +181,28 @@ class Meta:
 
     def __str__(self):
         return '{} MiB ({} MHz)'.format(self.size, self.speed)
+
+
+class FibreChannelCard(Component):
+    firmware_version = models.CharField(
+        verbose_name=_('firmware version'), max_length=255, blank=True,
+        null=True,
+    )
+    speed = models.PositiveIntegerField(
+        verbose_name=_('speed'), choices=FibreChannelCardSpeed(),
+        default=FibreChannelCardSpeed.unknown.id,
+    )
+
+    # If you need PWWN (or any other *WWN), add a separate field for
+    # it instead of using/re-using this one.
+    wwn = NullableCharField(
+        verbose_name=_('WWN'), max_length=255, unique=True, null=True,
+        blank=True, default=None,
+    )
+
+    class Meta:
+        verbose_name = _('fibre channel card')
+        verbose_name_plural = _('fibre channel cards')
+
+    def __str__(self):
+        return '{}'.format(self.model)
```

Comment:
Right. I'm thinking about something like `return 'model: "{}", WWN: "{}"'.format(self.model_name, self.wwn)` -  those extra labels and double quotes are for the case when some of those fields (or both of them) would be empty, so we'd get something like `<FibreChannelCard: model: "", WWN: "">` instead of just `<FibreChannelCard:  >`. What do you think..?